source/val/Function.cpp

```diff
@@ -215,6 +216,21 @@ void Function::RegisterBlockEnd(vector<uint32_t> next_list,
     next_blocks.push_back(&inserted_block->second);
   }
 
+  if (current_block_->is_type(kBlockTypeLoop)) {
+    // For each loop header, record the set of its successors, and include
+    // its continue target if the continue target is not the loop header
+    // itself.
+    std::vector<BasicBlock*>& next_blocks_plus_continue_target =
+        loop_header_successors_plus_continue_target_map_[current_block_];
+    next_blocks_plus_continue_target = next_blocks;
+    // If this block is marked as Loop-type,  then the continue construct is
+    // the most recently created CFG construct.
+    auto continue_target = cfg_constructs_.back().entry_block();
```

Comment:
Right.  But at this point I haven't got the Construct for this header block.  And Function doesn't have a map where I can look that up quickly.  I agree what I've got is a bit dodgy since it imposes a coupling between the implementations of the two methods. 

I plan to add a map in Function to find the construct from its entry block.